docs/developer-guide.md

```diff
@@ -0,0 +1,66 @@
+# Developer Getting Started Guide
+
+## Pre-requisites
+
+Start w/the following setup guide:
+
+[Setup-Guide-for-Kubernetes-Developers](https://developer.ibm.com/articles/setup-guide-for-kubernetes-developers/)
+
+**Skip the `kubernetes` steps from the guide above if you don't plan on doing any development in the `kubernetes` or `crictl` projects.**
+
+---
+`cri-tools` need the following programs installed in a system in order to work correctly:
+
+- [Go](https://golang.org/doc/install)
+- [Docker](https://docs.docker.com/get-docker/)
+- [Build tools](https://github.com/containerd/cri#install-dependencies)
+
+_**Important Note**: `Go` modules follow a similar structure to projects in Github, i.e. when installing and trying to use `cri-tools` as a local module, it should be installed under a folder structure as follows `go/src/github.com/kubernetes-sigs/cri-tools`. `k8s.io` is the only exception which does not follow this structure for legacy reasons._
+
+After following the setup guides from above, ensure the following:
+
+- There is a `go` directory is within your `$HOME` directory and the `go` binary is part of your environments `$PATH`
+
+  ```bash
+    $ ls $HOME
+    go
+
+    $ echo $PATH | grep go
+    PATH=$PATH:/usr/local/go/bin:$HOME/go/bin # $PATH here refers to truncated version of additional `env` paths that are unrelated this guide / setup
+  ```
+
+## Build & install `containerd/cri`
+
+If you followed the guide so far you may likely already have `containerd` installed, but we will need to install the latest development version.
+
+Running the development versions of `containerd` is not required if you do not plan on developing for `crictl`, `containerd` or `kubernetes`, you can instead install the latest packaged version for your given OS... i.e. for `Ubuntu` run `apt install contianerd`
+
+This guide will use `containerd/cri` as the container runtime. You may use `cri-o` instead but you may need some additional steps to ensure it works. You can also follow the `cri-o` [install guide](https://github.com/cri-o/cri-o/blob/master/install.md#install-packaged-versions-of-cri-o) if you prefer to use the packaged version.
+
+- Follow the [`containerd/cri` dev guide](https://github.com/containerd/cri#getting-started-for-developers) to `make` and install `containerd/cri` runtime.
+
+## Build & install `cri-tools`
+
+You can build and install `cri-tools` following the steps below:
```

Comment:
You can build and install `cri-tools` following the steps below: --> You can build and install `cri-tools` as follows: